yt/utilities/input_validator.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+
+from yt import YTQuantity, YTArray
+from yt.data_objects.static_output import Dataset
+from yt.utilities.exceptions import YTInvalidArgumentType
+
+def input_validator(_type_name):
+    def validator(func):
+        def wrapper(*args, **kwargs):
```

Comment:
You should probably use https://docs.python.org/3/library/functools.html#functools.wraps so that the decorated function has the right signature.

Reply:
👍

Reply:
thank you, done!